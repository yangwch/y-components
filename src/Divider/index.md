**Divider**

```tsx
import React from 'react';

import { Divider } from 'y-components';

const Paragraph = ({ children }) => <p style={{ margin: 0, padding: 0 }}>{children}</p>;

export default () => {
  return (
    <div>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
      <Divider />
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
    </div>
  );
};
```

**Inner Text**

```tsx
import React from 'react';

import { Divider } from 'y-components';

const Paragraph = ({ children }) => <p style={{ margin: 0, padding: 0 }}>{children}</p>;

export default () => {
  return (
    <div>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
      <Divider>Text Left</Divider>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
      <Divider align="center">Text Center</Divider>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
      <Divider align="right">Text Right</Divider>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi
        ista probare, quae sunt a te dicta? Refert tamen, quo modo.
      </Paragraph>
    </div>
  );
};
```

**Orientation**

```tsx
import React from 'react';

import { Divider, Button } from 'y-components';

export default () => {
  return (
    <div>
      <div> Vertical & Custom Style</div>
      <div style={{ fontSize: '22px' }}>
        <span>Left</span>
        <Divider type="vertical" style={{ height: 25 }} />
        <span> Center</span>
        <Divider type="vertical" style={{ borderColor: 'red', margin: '0 30px' }} />
        <span style={{ color: 'red' }}>Right</span>
      </div>
    </div>
  );
};
```
