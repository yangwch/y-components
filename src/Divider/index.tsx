import classNames from 'classnames';
import React, { CSSProperties, HTMLAttributes } from 'react';
import { settings } from '../utils/global';
import './style/index.less';

const prefix = settings.prefix;

export type DividerType = 'horizontal' | 'vertical';

export type Alignment = 'left' | 'center' | 'right'

interface DividerProps extends HTMLAttributes<HTMLDivElement> {
  type?: DividerType;
  children?: React.ReactNode;
  align?: Alignment
}
function Divider(props: DividerProps) {
  const { type, children, align, ...rest } = props;

  const direction: string = type || 'horizontal'
  const cls = classNames(
    `${prefix}-divider`,
    `${prefix}-divider-${direction}`,
    {
      [`${prefix}-divider-${align}`]: !!align

    }
  );
  return (
    <div className={cls} {...rest}>
      <span className="text">{children}</span>
    </div>
  );
}

export default Divider;
