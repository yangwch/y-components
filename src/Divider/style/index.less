@import '../../style/themes/index.less';
// @import '../../style/mixins/reset.less';

@divider-cls: ~'@{prefix}-divider';

.@{divider-cls} {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1;
  list-style: none;
  font-feature-settings: "tnum";
  border-top: 1px solid @divider-color;
  position: relative;

  &-horizontal {
    min-width: 100%;
    margin-top: 1em;

    .text {
      transform: translate(0, -50%);
      display: inline-block;
      margin: 0 @divider-text-margin;
      padding: 0 @divider-text-padding;
      background: @component-background;
    }
  }

  &-vertical {
    position: relative;
    display: inline-block;
    top: -.06em;
    height: .9em;
    margin: 0 @divider-vertical-gutter;
    vertical-align: middle;
    border-left: 1px solid @divider-color;
    .text { display: none; }
  }

  &-center {
    text-align: center;
  }

  &-right {
    text-align: right;
  }
}