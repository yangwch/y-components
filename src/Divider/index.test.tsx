import React from 'react';
import { render } from '@testing-library/react';
import { mountTest } from '../../test/shared/mountTest';

import Divider from './index';

describe('Divider', () => {
  mountTest(() => <Divider />)
  mountTest(() => <Divider>test</Divider>)
  mountTest(() => <Divider type="vertical" />)
  it('render correctly', () => {
    const testFn = () => render(<Divider>Test</Divider>);
    expect(testFn).not.toThrowError();
  
    const { container } = testFn();
    expect(container.firstChild).toMatchSnapshot();
  });
})
