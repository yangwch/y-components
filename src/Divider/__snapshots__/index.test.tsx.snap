// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  class="yy-divider yy-divider-horizontal"
>
  <span
    class="text"
  >
    Test
  </span>
</div>
`;
